x-pack/test/api_integration/apis/cloud/set_cloud_data_route.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function ({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+
+  describe('POST /internal/cloud/solution', () => {
+    it('set solution data', async () => {
+      await supertest
+        .post('/internal/cloud/solution')
+        .set('kbn-xsrf', 'xxx')
+        .set('x-elastic-internal-origin', 'cloud')
+        .set('elastic-api-version', '1')
+        .send({
+          onboardingData: {
+            solutionType: 'search',
+            token: 'connectors',
```

Comment:
how confident are we that token won't expose sensitive data?

Reply:
we are super confident ;)